code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -130,8 +146,14 @@ public HandlerResult doWithLock() {
                         try {
                             objectProcessManager.scheduleStandardProcess(StandardProcess.REMOVE, instance, data);
                         } catch (ProcessCancelException e) {
-                            data.put(REMOVE_OPTION, true);
-                            objectProcessManager.scheduleProcessInstance(PROCESS_STOP, instance, data);
+                            if (STATE_STOPPING.equals(state)) {
+                                // handle docker forced stop and remove
+                                instance = resourceMonitor.waitForNotTransitioning(instance);
+                                objectProcessManager.scheduleStandardProcess(StandardProcess.REMOVE, instance, data);
+                            } else {
+                                data.put(REMOVE_OPTION, true);
```

Comment:
Not required for this PR, but just a hint: we can chain processes using ProcessUtils.chainInData method